picky/src/x509/pkcs7/authenticode.rs

```diff
@@ -991,15 +990,15 @@ impl<'a> AuthenticodeValidator<'a> {
     }
 }
 
-fn h_check_eku_code_signing(certificates: &[Cert]) -> AuthenticodeResult<()> {
-    if certificates
-        .iter()
+fn h_check_eku_code_signing(certificates: &[Cert], signing_certificate: &Cert) -> AuthenticodeResult<()> {
+    let certificates_iter = SignatureCertificatesIterator::new(signing_certificate, certificates.iter());
+
+    if certificates_iter
         .flat_map(|cert| cert.extensions().iter())
         .any(|extension| matches!(extension.extn_value(), ExtensionView::ExtendedKeyUsage(_)))
     {
-        let signing_cert = certificates.first().ok_or(AuthenticodeError::NoCertificates)?;
-        if !signing_cert
-            .extensions()
+        let signing_certificate_extensions = signing_certificate.extensions();
+        if !signing_certificate_extensions
```

Comment:
Not a big deal, but there is probably little value in declaring an intermediate variable in this case.